pageserver/src/tenant/mgr.rs

```diff
@@ -489,7 +501,7 @@ where
 async fn remove_tenant_from_memory<V, F>(
     tenant_id: TenantId,
     tenant_cleanup: F,
-) -> anyhow::Result<V>
+) -> anyhow::Result<V, TenantStateError>
```

Comment:
I think there was already a comment about this, but using anyhow::Result like this is confusing, just use Result:

```suggestion
) -> Result<V, TenantStateError>
```